File: src/main/java/itmo/is/project/mapper/module/production/ProductionModuleMapper.java
package itmo.is.project.mapper.module.production;

import itmo.is.project.dto.module.production.ProductionModuleDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.module.production.ProductionModule;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring", uses = {ProductionModuleBlueprintMapper.class})
public interface ProductionModuleMapper extends EntityMapper<ProductionModuleDto, ProductionModule> {

    @Override
    @Mapping(target = "productionModuleBlueprintDto", source = "blueprint")
    ProductionModuleDto toDto(ProductionModule entity);

    @Override
    @Mapping(target = "blueprint", source = "productionModuleBlueprintDto")
    ProductionModule toEntity(ProductionModuleDto dto);
}

File: src/main/kotlin/itmo/isproject/production/mapper/ProductionModuleMapper.kt
package itmo.isproject.production.mapper

import itmo.isproject.production.dto.ProductionModuleDto
import itmo.isproject.shared.common.mapper.EntityMapper
import itmo.isproject.production.model.ProductionModule
import org.mapstruct.Mapper

@Mapper(uses = [ProductionModuleBlueprintMapper::class])
interface ProductionModuleMapper : EntityMapper<ProductionModuleDto, ProductionModule>
